src/hooks/useFetchMovies.ts

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+import { MovieInfo } from '../components/movie-tile/movie-tyle.types';
+
+export const useFetchMovies = (
+  searchQuery: string,
+  searchBy: 'title' | 'genres' = 'title',
+  sortBy: 'title' | 'release_date' = 'release_date',
+  genreFilter?: string,
+) => {
+  const [movies, setMovies] = useState<MovieInfo[]>([]);
+
+  useEffect(() => {
+    const fetchMovies = async () => {
```

Comment:
Please, check what race conditions are, AbortController, and how to use AbortController in order to deal with race conditions in React.
For example: https://www.youtube.com/watch?v=VdAlFWBUGV4